baseplate/observers/prometheus.py

```diff
@@ -253,3 +254,14 @@ def on_child_span_created(self, span: Span) -> None:
             observer = PrometheusClientSpanObserver()
 
         span.register(observer)
+
+
+class PrometheusGenericSpanObserver(PrometheusLocalSpanObserver):
+    def __init__(self, span_name: Optional[str] = None) -> None:
+        super(PrometheusGenericSpanObserver, self).__init__(span_name)
+        self.metrics: PrometheusGenericSpanMetrics = PrometheusGenericSpanMetrics(span_name)
```

Comment:
That's going to create a new metric name per span. You don't want that. I think you would want like a common name `generic_server_` or smth, and the having the span being one of the labels.

Reply:
For the test case I had this actually ends up being `None`, but I wanted a way to differentiate different generic spans without just using tags

Reply:
But you pass span_name as an argument that you then set as prefix I believe?
My concern now would be that the variables are using a None prefix whatever you do because they're created at the class definition level, not the instance. Which means that this would never work?

Reply:
@Xaelias Thanks for calling that out. We definitely don't want service names leaking into metric names. That's the job of Prometheus discovery labels. (i.e. Kubernetes namespace, etc, labels).

Reply:
Discussed in slack, there are certain metrics that look a heck of a lot like service names but aren't actually. Those metrics (like the ones below) are fine.